Remove some additional items from resource dropdown

spadgett opened the pull request:
Remove items that can't be listed and a few well-known items that
appear in different API groups.

This also improves the resource icon abbreviation to use the capital
letters from a resource instead of just the first 3 letters (which are
often repeated for many resources).

/cc @alecmerdler @rhamilto 

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1609729
Fixes  https://bugzilla.redhat.com/show_bug.cgi?id=1609733
Fixes CONSOLE-651

<img width="327" alt="search openshift origin 2018-07-26 20-03-11" src="https://user-images.githubusercontent.com/1167259/43294733-f1e8cfa8-910e-11e8-9c4a-f514594a94f9.png">

benjaminapetersen commented on frontend/public/components/resource-dropdown.tsx:
```diff
@@ -22,9 +31,20 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = props => {
+        return false;
+      }
+
+      // Only show preferred version for resources in the same API group.
```
Do you want to redeclare the functions in every iteration of filter?  Its prob not a big deal as its a set of a known size.

benjaminapetersen replied:
other than that i think its an lgtm, though the tests are complaining 😦

spadgett replied:
Good eye... It's a little difficult to pull out since it uses vars from the closure. Also not new to this PR.

Do you mind if I punt on that? The list should never be too large

spadgett commented:
ALM tests fixed in #326

spadgett commented:
Updated with a fix for CONSOLE-651 and removed some additional duplicates on OpenShift clusters.

Also fixes https://bugzilla.redhat.com/show_bug.cgi?id=1609733

@benjaminapetersen PTAL
